tokio-timer/src/lib.rs

```diff
@@ -64,6 +64,11 @@ pub fn sleep(duration: Duration) -> Delay {
     Delay::new(Instant::now() + duration)
 }
 
+/// Create a Future that completes at `deadline`.
+pub fn delay(deadline: Instant) -> Delay {
```

Comment:
Should this be tagged with `#[inline]`?

Reply:
The compiler should be smart enough about that.

Reply:
Does rust do any cross-crate inlining of monomorphic functions that aren't explicitly marked whatsoever right now?

Reply:
According to https://github.com/rust-random/rand/issues/472#issue-325644602, it only inlines monomorphic functions only when LTO is used or it's marked explicitly. However, `Delay::new` wasn't marked inline before, so nothing has really changed, the compiler can inline `Delay::new` into `delay` inside the crate and to the outside the situation will stay the same.

I created a binary crate outside of `tokio`, included `tokio-timer`, called `delay`, compiled in release mode and `objdump` showed that the call to `delay` is not inlined, but `Delay::new` was inlined into `delay`. So the situation should stay as it was before with this PR.

Reply:
We can change this in a follow up PR